common/system/tasks/packet_encoder.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "board_config.h"
+#include "radio_packet_types.h"
+#include "ti_fec.h"
+
+class FSKPacketRadioEncoder {
+ public:
+  int Encode(RadioDecodedPacket_s& input, RadioOTAPayload_s& output);
+  int Decode(RadioOTAPayload_s& input, RadioDecodedPacket_s& output);
+
+ private:
+  FecEncoder encoder;
+  FecDecoder decoder;
+};
+
+class PassthroughRadioEncoder {
```

Comment:
Probably worth making an interface if there are multiple possible packet encoders.

Reply:
Can we use concepts?

Reply:
How would you propose doing this without paying the interface overhead cost? just make an interface but still use instances of the child classes in code?